lib/stores/FileStore.js

```diff
@@ -92,78 +75,73 @@ class FileStore extends DataStore {
 
     /** Get file from filesystem
      *
-     * @param {string} file_id    Name of the file
+     * @param {string} file_id  Name of the file
      *
      * @return {stream.Readable}
      */
     read(file_id) {
-        const path = `${this.directory}/${file_id}`;
-        return fs.createReadStream(path);
+        return fs.createReadStream(path.join(this.directory, file_id));
     }
 
     /**
      * Deletes a file.
      *
-     * @param  {object} req http.incomingMessage
+     * @param {string} file_id  Name of the file
      * @return {Promise}
      */
-    remove(req) {
+    remove(file_id) {
         return new Promise((resolve, reject) => {
-            const file_id = req.file_id;
-            return fs.unlink(`${this.directory}/${file_id}`, (err, fd) => {
+            return fs.unlink(`${this.directory}/${file_id}`, async(err, fd) => {
                 if (err) {
-                    console.warn('[FileStore] delete: Error', err);
-                    return reject(ERRORS.FILE_NOT_FOUND);
+                    log('[FileStore] delete: Error', err);
+                    reject(ERRORS.FILE_NOT_FOUND);
+                    return;
+                }
+
+                try {
+                    resolve(await this.configstore.delete(file_id));
+                }
+                catch (error) {
+                    reject(error);
                 }
-                this.emit(EVENTS.EVENT_FILE_DELETED, { file_id });
-                this.configstore.delete(file_id);
-                return resolve();
             });
         });
     }
 
     /**
      * Write to the file, starting at the provided offset
      *
-     * @param  {object} req http.incomingMessage
-     * @param  {string} file_id   Name of file
-     * @param  {integer} offset     starting offset
+     * @param {object} readable stream.Readable
+     * @param {string} file_id Name of file
+     * @param {integer} offset starting offset
      * @return {Promise}
      */
-    write(req, file_id, offset) {
-        return new Promise((resolve, reject) => {
-            const path = `${this.directory}/${file_id}`;
-            const options = {
-                flags: 'r+',
-                start: offset,
-            };
-
-            const write_stream = fs.createWriteStream(path, options);
-            if (!write_stream || req.destroyed) {
-                reject(ERRORS.FILE_WRITE_ERROR);
-                return;
-            }
+    async write(readable, file_id, offset) {
+        const writeable = fs.createWriteStream(path.join(this.directory, file_id), {
+            flags: 'r+',
+            start: offset,
+        });
 
-            let new_offset = 0;
-            req.on('data', (buffer) => {
-                new_offset += buffer.length;
-            });
+        let new_offset = 0;
+        readable.pause();
```

Comment:
why the `pause`/`resume` in the same function?

Reply:
Stream will switch in flowing mode when `data` listener is attached. This is to prevent any data loss before `stream.pipeline(...)` is called later on. But after some more thought I guess this is not needed as no data can be emited before execution reaches event loop.

Reply:
Maybe we can also get rid of the data listener and simply stat the file again to see the new offset. Not sure if that's worth the effort though.